enhancements/console/samples.md

```diff
@@ -0,0 +1,415 @@
+3. `description` is required, has no maximum length, and is a README.md-like content for
+   additional information, links, pre and post conditions and instrucations.
+   It will be rendered as **Markdown** so that it can contain line breaks, links, and other simple formatting.
+4. `icon` is an optional base64 encoded image and shown beside the sample title.
+   The format must follow the [data: URL format](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URLs).
+
+   `data:[<mediatype>][;base64],<base64 encoded image>`
+
+   For example:
+
+   `data:image;base64,` plus the base64 encoded SVG image.
+
+   Vector images can also be used. SVG icons must start with:
+
+   `data:image/svg+xml;base64,` plus the base64 encoded SVG image.
+
+   All sample catalog icons will be shown on a white background (also when the dark theme is used).
+   The web console ensures that different aspect radios work correctly.
+   Currently, the surface of the icon is at most 40x100px.
+
+5. `type` is an optional label to group multiple samples. It can have a maximum length of **20** characters.
+
+   Currently, the type is shown a badge on the sample card tile in the top right corner (see screenshot above).
+
+   Recommendation is a **singular term** that groups samples like "Builder Image", "Devfile" or "Serverless Function".
+
+6. `provider` is an optional label to honor who provides the sample. It can have a maximum length of **20** characters.
+
+   A provider can be a company like "Red Hat" or an organization like "CNCF" or "Knative".
+
+   Currently, the provider is only shown on the sample card tile below the title with the prefix "Provided by " (see screenshot above).
+
+7. `tags` are optional.
+
+   Tags can be used to search for samples in the samples catalog.
+
+   They will be also displayed on the samples details page.
+
+### Source types
+
+The `source` attribute is an union type and will support only two `type`s in 4.14:
+
+1. `GitImport` provides information to import an application from a Git repository.
+2. `ContainerImport` provides information to import a container image.
+
+Other types will be ignored by the web console so that it is possible that operators install `type`s and downgrade the cluster when other source types are supported in the future.
+
+#### `GitImport`
+
+All previously supported samples opens a simplified "Import from Git" flow and prefill the Git repository.
+
+When a `GitImport` sample is selected, the same form will be opened and prefilled.
+
+1. The Git `url` is required.
+2. An optional Git `revision` to clone a specific branch, tag or commit SHA, etc.
+3. An optional `contextDir`, a subfolder in the repo to build the component.
+
+There is no other attribute supported at the moment. It supports only public Git repositories.
+
+```yaml
+  source:
+    type: GitImport
+    gitImport:
+      url: https://github.com/jboss-openshift/openshift-quickstarts
+      revision: master # optional
```

Comment:
nitpick: I'd use `main` as the example here. It's the new default and we're trying to get away from the term `master`.